src/signer/wallet.py

```diff
@@ -76,36 +76,33 @@ def set_sig_hash(self, sig_hash):
     def set_current_spend_request_id(self, current_spend_request_id):
         self.current_spend_request_id = current_spend_request_id 
 
-    def get_private_key_tweaked(self):
-        if self.cmap != None:
-            # TODO this is all bip32 stuff
-            # TODO index is hardcoded at 1
-            # index = 1
-            # pk = self.get_pubkey_at_index(index).hex()
-            # TODO hardcoded pk, get from class variable
-            # prv = self.get_root_hd_node().get_privkey_from_path(f"m/{index}")
-            # print("prv", prv)
-            # private_key = ECKey().set(prv)
-
-            pk = self.public_key.get_bytes().hex()
-            private_key = self.private_key
-            tweaked_key = private_key * self.cmap[pk]
-
-            if self.pubkey_agg.get_y() % 2 != 0:
-                tweaked_key.negate()
-                self.pubkey_agg.negate()
-
-            return tweaked_key
-        return None
-
-    def sign_with_current_context(self, nonce: str):
+    def get_private_key_tweaked(self, address_index: int):
+        if self.cmap == None:
+            return None
+        pk = self.get_pubkey_at_index(address_index).hex()
+        prv = self.get_root_hd_node().get_privkey_from_path(f"m/{address_index}")
+        print("prv", prv)
+        private_key = ECKey().set(prv)
+
+        # pk = self.public_key.get_bytes().hex()
+        # private_key = self.private_key
+        tweaked_key = private_key * self.cmap[pk]
```

Comment:
Just a thought - is it okay to multiply values like this? I would think for Schnorr it's fine since it has those nice linear properties that ECDSA doesn't, but it could be worth double checking if this is where the negation bug is coming from.